enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+
+* `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --channel alpha --namespace memcached-system --install-mode SingleNamespace=default`
+
+This is how the presence of an OperatorGroup in the
+namespace affects `run bundle-upgrade` behavior. 
+
+* Given an operator that supports install modes `OwnNamespace` and
+  `SingleNamespace` being upgraded into a namespace with *no* OperatorGroup.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0`
+    * Upgrade into the default namespace configured in the `KUBECONFIG` context. An OperatorGroup would be created targeting the same install namespace.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --namespace memcached-system --install-mode SingleNamespace=default`
+      * Upgrade into namespace `memcached-system`. An OperatorGroup would be created targeting the `default` namespace.
+* Given an operator that supports install modes `AllNamespaces`, `OwnNamespace`, and `SingleNamespace`, being upgraded into a namespace with an existing OperatorGroup that targets all namespaces:
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0` 
+    * Upgrade into the default namespace configured in the `KUBECONFIG` context. The operator would watch all namespaces.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --namespace memcached-system --install-mode SingleNamespace=default`
+    * Upgrade into namespace `memcached-system`. The operator would watch namespace `default`.
+
+#### Simple Example
+
+Here's a simple example with all the required flags.
+
+```
+operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0
+```
+
+Below is the workflow when the above command is invoked:
+
+ 1. Create pod using `--index-image`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported index image. Also capture the pod's logs for debugging.
+ 1. Create GRPC CatalogSource, pointing to `<pod-name>.<namespace>:50051`
+ 1. If an OperatorGroup doesn't exist, create one and default to `AllNamespaces` as `--install-mode` was not provided
+ 1. Create Subscription
```

Comment:
nit: should these list numbers be increasing?

Reply:
in markdown, this will render a monotonically increasing sequence in the output automatically. The reason this is also allowed because when we make list additions, we don't have to keep tweaking the numbers. :)
Although I agree that it's confusing when we don't know about this feature